### ISSUE TITLE
Dependency lookup somehow broken when managing a react-static site with Lerna

### ISSUE THREAD

**Issue by chadoh** [nozzle/react-static#74]
**TL;DR** dependency lookup in a react-static site somehow breaks when one of the dependencies is a symlink generated by Lerna.

This issue reproduced and documented [here](https://github.com/chadoh/lerna-repo/commit/73add6b3b938fc817facc0e680955ea597c29523).

---

I'm managing a set of components and sites with Lerna. The relevant libs/apps:

* `camera` – a component used by the two apps which is not yet published on NPM
* `webapp` – a site generated with `create-react-app` which relies on `camera`. Running `lerna bootstrap` in the monorepo root creates a symlink in the `node_modules` folder of this app, and running the app with `yarn start` works correctly. All dependencies are resolved.
* `static` – a static site generated with`react-static`, which also relies on `camera`. Running `lerna bootstrap` in the monorepo root creates a symlink in the `node_modules` folder of this site, _but running the site with `yarn start` fails!!_ It says things like `Module not found: Can't resolve 'gifshot' in '/path/to/lerna-repo/packages/camera/lib/Camera'`. 

How does a `react-static` site resolve dependencies? Why and how would it be different from how a `create-react-app` site does it?

**Comment by chadoh** [comment]
Could this be related to HMR, Hot Module Reloading? Is there some way I can turn it off to check?

**Comment by chadoh** [comment]
(I really don't need HMR anyhow)

**Comment by chadoh** [comment]
I tried my best to remove HMR from the webpack dev server configuration within my local repo, but the problem persists.

**Comment by chadoh** [comment]
I tried switching to Yarn Workspaces per [their blog post](https://yarnpkg.com/blog/2017/08/02/introducing-workspaces/).

This causes even more failures then when using `react-static` with regular `lerna`. However, an app created with `create-react-app` continues to work fine. See [this commit](https://github.com/chadoh/lerna-repo/commit/7895690cad303aea4ccc752a31ad38e9900f6473) for more details, and [this commit](https://github.com/chadoh/lerna-repo/commit/73add6b3b938fc817facc0e680955ea597c29523) for instructions on how to reproduce the issues.

How is `react-static` looking up dependencies, and why is it so broken?

**Comment by tannerlinsley** [comment]
React-Static uses standard node module resolution for everything it imports, so I'm not sure why you are experiencing this problem. I know lerna is difficult to integrate with CLI's that assume a project root, and that could be the case here.  There are some things that use the CWD to resolve things like `static.config.js` and your main entry file. 

Regardless of how we move forward on this, you're going to want to upgrade to at least version 3.0.3, and perform any necessary migration as outlined in the Changelog. Then we can take a closer look.

**Comment by sammkj** [comment]
I tested it in my monorepo and got the follow errors:

```
react-static create

? What should we name this project? world
? Select a template below... basic
=> Creating new react-static project...
=> Installing dependencies...

cd world

yarn start

$ react-static start

=> Copying public directory...
=> [✓] Public directory copied: 1.228ms
=> Building Routes...
=> [✓] Routes Built: 385.992ms
=> Building App Bundle...
=> [✓] App serving at http://localhost:3000
Failed to rebuild.
/home/sammkj/Desktop/platform/node_modules/react-dev-utils/formatWebpackMessages.js
Module not found: Can't resolve 'chalk' in '/home/sammkj/Desktop/platform/node_modules/react-dev-utils'

/home/sammkj/Desktop/platform/node_modules/react-hot-loader/lib/patch.dev.js
Module not found: Can't resolve 'global' in '/home/sammkj/Desktop/platform/node_modules/react-hot-loader/lib'

./dist/react-static-routes.js
Module not found: Can't resolve 'react' in '/home/sammkj/Desktop/platform/apps/web/world/dist'

./src/index.js
Module not found: Can't resolve 'react' in '/home/sammkj/Desktop/platform/apps/web/world/src'

./src/App.js
Module not found: Can't resolve 'react' in '/home/sammkj/Desktop/platform/apps/web/world/src'

./src/containers/About.js
Module not found: Can't resolve 'react' in '/home/sammkj/Desktop/platform/apps/web/world/src/containers'

./src/containers/Blog.js
Module not found: Can't resolve 'react' in '/home/sammkj/Desktop/platform/apps/web/world/src/containers'

./src/containers/404.js
Module not found: Can't resolve 'react' in '/home/sammkj/Desktop/platform/apps/web/world/src/containers'

./src/containers/Post.js
Module not found: Can't resolve 'react' in '/home/sammkj/Desktop/platform/apps/web/world/src/containers'

./src/containers/Home.js
Module not found: Can't resolve 'react' in '/home/sammkj/Desktop/platform/apps/web/world/src/containers'

/home/sammkj/Desktop/platform/node_modules/react-hot-loader/lib/patch.dev.js
Module not found: Can't resolve 'react' in '/home/sammkj/Desktop/platform/node_modules/react-hot-loader/lib'

./src/index.js
Module not found: Can't resolve 'react-dom' in '/home/sammkj/Desktop/platform/apps/web/world/src'

/home/sammkj/Desktop/platform/node_modules/react-dev-utils/webpackHotDevClient.js
Module not found: Can't resolve 'react-error-overlay' in '/home/sammkj/Desktop/platform/node_modules/react-dev-utils'

./src/index.js
Module not found: Can't resolve 'react-hot-loader' in '/home/sammkj/Desktop/platform/apps/web/world/src'

/home/sammkj/Desktop/platform/node_modules/react-hot-loader/lib/patch.dev.js
Module not found: Can't resolve 'react-proxy' in '/home/sammkj/Desktop/platform/node_modules/react-hot-loader/lib'

./dist/react-static-routes.js
Module not found: Can't resolve 'react-router-dom' in '/home/sammkj/Desktop/platform/apps/web/world/dist'

./src/App.js
Module not found: Can't resolve 'react-static' in '/home/sammkj/Desktop/platform/apps/web/world/src'

./src/containers/Post.js
Module not found: Can't resolve 'react-static' in '/home/sammkj/Desktop/platform/apps/web/world/src/containers'

./src/containers/Blog.js
Module not found: Can't resolve 'react-static' in '/home/sammkj/Desktop/platform/apps/web/world/src/containers'

./src/containers/Home.js
Module not found: Can't resolve 'react-static' in '/home/sammkj/Desktop/platform/apps/web/world/src/containers'

/home/sammkj/Desktop/platform/node_modules/react-dev-utils/webpackHotDevClient.js
Module not found: Can't resolve 'sockjs-client' in '/home/sammkj/Desktop/platform/node_modules/react-dev-utils'

/home/sammkj/Desktop/platform/node_modules/react-dev-utils/webpackHotDevClient.js
Module not found: Can't resolve 'strip-ansi' in '/home/sammkj/Desktop/platform/node_modules/react-dev-utils'
```

I verified that those packages were in `node_modules`. Not sure why it threw `Module not found`.

**Comment by tannerlinsley** [comment]
We just release an alpha `4.0.0-alpha-0`, care to try that out?

**Comment by tannerlinsley** [comment]
Marking this issue as stale. If you'd like to pursue it still, reopen :)

**Comment by cooperka** [comment]
I had similar issues with Lerna using react-static v5.9.6 -- I submitted https://github.com/nozzle/react-static/pull/627 as a solution.
